### PR TITLE
Battle Royale requires +ADMIN instead of +FUN

### PR DESCRIPTION
**Pull request by Ivniinvi**
## About The Pull Request
title

## Why It's Good For The Game
When the only admin online is a moderator and has to delay the round for 15 minutes to address tickets, it gets kinda boring for ghosts. This lets the 60+ ghosts just sitting there be not-bored.

## Changelog
:cl:
admin: Battle Royale requires +ADMIN instead of +FUN after the round ends.
/:cl:

### CODE REVIEW

**Comment by OceanFish1** [comment]
Just do your job faster loool

**Comment by PowerfulBacon** [comment]
Fun is not allowed

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Not sure how I feel about all admins having access to it. If any admin can do it, then normal admins should only be able to do it post round end.

**Comment by Ivniinvi** [comment]
Battle Royale now is back to R_FUN normally, and only switches to R_ADMIN after the round ends.

**Comment by ike709** [comment]
Update the CL

**Comment by Ivniinvi** [comment]
woops, done.

**Review by PowerfulBacon — DISMISSED** [review]

File: code/modules/admin/battle_royale.dm [hunk 1]
```diff
@@ -116,7 +116,7 @@ GLOBAL_DATUM(battle_royale, /datum/battle_royale_controller)
 /client/proc/battle_royale()
 	set name = "Battle Royale"
 	set category = "Fun"
-	if(!check_rights(R_FUN))
+	if(!(check_rights(R_FUN) || (check_rights(R_ADMIN) && SSticker.current_state == GAME_STATE_FINISHED)))
 		to_chat(src, "<span class='warning'>You do not have permission to do that!</span>")
```
Comment (ike709) [inline code/modules/admin/battle_royale.dm]: Should clarify for admins that they can use it after roundend

File: code/modules/admin/battle_royale.dm [hunk 2]
```diff
@@ -117,7 +117,7 @@ GLOBAL_DATUM(battle_royale, /datum/battle_royale_controller)
 	set name = "Battle Royale"
 	set category = "Fun"
 	if(!(check_rights(R_FUN) || (check_rights(R_ADMIN) && SSticker.current_state == GAME_STATE_FINISHED)))
-		to_chat(src, "<span class='warning'>You do not have permission to do that!</span>")
+		to_chat(src, "<span class='warning'>You do not have permission to do that! (If you don't have +FUN, wait until the round is over.)</span>")
```
Comment (ike709) [inline code/modules/admin/battle_royale.dm]:
```suggestion
		to_chat(src, "<span class='warning'>You do not have permission to do that! (If you don't have +FUN, wait until the round is over then you can trigger it.)</span>")
```

File: code/modules/admin/battle_royale.dm [hunk 3]
```diff
@@ -116,8 +116,8 @@ GLOBAL_DATUM(battle_royale, /datum/battle_royale_controller)
 /client/proc/battle_royale()
 	set name = "Battle Royale"
 	set category = "Fun"
```
Comment (ike709) [inline code/modules/admin/battle_royale.dm]: Can admins actually see the `Fun` tab if they don't have `+FUN`? If not, the category needs to be changed or they'll have an entire `Fun` table with nothing but this one verb they can't use until roundend on it.

Comment (Ivniinvi) [reply code/modules/admin/battle_royale.dm]: We can, there's three or four verbs in there for mods at the moment.

**Review by ike709 — DISMISSED** [review]

File: code/modules/admin/battle_royale.dm (hunk 3 above)
Comment (Ivniinvi) [reply code/modules/admin/battle_royale.dm]: Never mind, we don't have the fun tab, moving it to Adminbus tab.

Comment (Ivniinvi) [reply code/modules/admin/battle_royale.dm]: Fixed

**Review by PowerfulBacon — APPROVED** [review]

**Comment by francinum** [comment]
IMO the solution here is not to delay the round by 15 minutes often enough for this to be necessary.

**Comment by Ivniinvi** [comment]
> IMO the solution here is not to delay the round by 15 minutes often enough for this to be necessary.

If there were more admins on at midnight eastern time, I'd be all for that, but at the moment it's hard enough to deal with eight tickets at the same time without dead chat and OOC telling me how bored they are.

**Pull request merged**